cpp/oneapi/dal/algo/rbf_kernel/backend/gpu/compute_kernel_dense_dpc.cpp

```diff
@@ -15,60 +15,114 @@
+            const Float arg = sycl::fmax((sqr_x_i + sqr_y_j + res_rbf_ij) * coeff, threshold);
+
+            res_rbf_ptr[i * ld + j] = sycl::exp(arg);
+        });
+    });
+
+    return compute_rbf_event;
+}
 
 template <typename Float>
-static result_t call_daal_kernel(const context_gpu& ctx,
-                                 const descriptor_t& desc,
-                                 const table& x,
-                                 const table& y) {
+static result_t compute(const context_gpu& ctx, const descriptor_t& desc, const input_t& input) {
+    const auto x = input.get_x();
+    const auto y = input.get_y();
+
     auto& queue = ctx.get_queue();
-    interop::execution_context_guard guard(queue);
 
     const int64_t row_count_x = x.get_row_count();
+    const int64_t col_count_x = x.get_column_count();
     const int64_t row_count_y = y.get_row_count();
+    const int64_t col_count_y = y.get_column_count();
 
+    ONEDAL_ASSERT(col_count_x == col_count_y);
     dal::detail::check_mul_overflow(row_count_x, row_count_y);
-    auto arr_values =
-        array<Float>::empty(queue, row_count_x * row_count_y, sycl::usm::alloc::device);
-
-    const auto daal_x = interop::convert_to_daal_table(queue, x);
-    const auto daal_y = interop::convert_to_daal_table(queue, y);
-    const auto daal_values =
-        interop::convert_to_daal_table(queue, arr_values, row_count_x, row_count_y);
-
-    daal_rbf_kernel::Parameter daal_parameter(desc.get_sigma());
-    daal_rbf_kernel_t<Float>().compute(daal_x.get(),
-                                       daal_y.get(),
-                                       daal_values.get(),
-                                       &daal_parameter);
-
-    return result_t().set_values(
-        dal::detail::homogen_table_builder{}.reset(arr_values, row_count_x, row_count_y).build());
-}
 
-template <typename Float>
-static result_t compute(const context_gpu& ctx, const descriptor_t& desc, const input_t& input) {
-    return call_daal_kernel<Float>(ctx, desc, input.get_x(), input.get_y());
+    const Float sigma = desc.get_sigma();
+    const Float coeff = static_cast<Float>(-0.5 / (sigma * sigma));
+
+    auto arr_x = row_accessor<const Float>(x).pull(queue, { 0, -1 }, sycl::usm::alloc::device);
+    const auto ndarray_x = pr::ndarray<Float, 2>::wrap(arr_x, { row_count_x, col_count_x });
+
+    auto arr_y = row_accessor<const Float>(y).pull(queue, { 0, -1 }, sycl::usm::alloc::device);
+    const auto ndarray_y = pr::ndarray<Float, 2>::wrap(arr_y, { row_count_y, col_count_y });
+
+    auto ndarray_res =
+        pr::ndarray<Float, 2>::empty(queue, { row_count_x, row_count_y }, sycl::usm::alloc::device);
+
+    auto [ndarray_sqr_x, sqr_x_zeros_event] =
+        pr::ndarray<Float, 1>::zeros(queue, { row_count_x }, sycl::usm::alloc::device);
+    auto [ndarray_sqr_y, sqr_y_zeros_event] =
+        pr::ndarray<Float, 1>::zeros(queue, { row_count_y }, sycl::usm::alloc::device);
```

Comment:
Not necessary fill with zeros, reduction will initialize the data, `ndarray::empty` is enough